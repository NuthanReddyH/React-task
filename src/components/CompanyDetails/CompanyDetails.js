import React, { Component } from "react";


class CompanyDetails extends Component {

  
  render() {
      console.log(this.props.location.state);
    return (
      <div>Details</div>
    );
  }
}

export default CompanyDetails;
