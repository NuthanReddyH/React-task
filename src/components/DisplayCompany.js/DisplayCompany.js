import React, { Component } from "react";
import {
  Card,
  CardHeader,
  CardFooter, CardBody,
} from "reactstrap";
import { companyDetails } from '../../mock.js';
import { Link } from 'react-router-dom'

class DisplayCompany extends Component {

renderCards() {
  return companyDetails.map((item,index) => {
        return (
          <Card key={index} className="cardDetails">
                <CardHeader>{item.companyName}</CardHeader>
                <CardBody>
                      <label><strong>Address : </strong></label> {item.address} <br/>
                      <label><strong>Revnue : </strong></label> {item.revenue}<br/>
                      <label><strong>Phone : </strong></label> {item.phone}
                </CardBody>
                <CardFooter><Link  to={{ pathname: '/details', state: { ...item} }}>Company Overview</Link></CardFooter>
          </Card>
        )
  })
}


  render() {
    return (
      <div className="ccontainer">
        <Card>
            <CardHeader>Companies</CardHeader>
           {this.renderCards()}
        </Card>
      </div>
    );
  }
}

export default DisplayCompany;
