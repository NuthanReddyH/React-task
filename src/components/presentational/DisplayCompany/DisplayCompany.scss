.errorMessage {
    position: absolute;
    left: 40%;
    top: 45%;
    transform: translate(-50%, -50%);
}