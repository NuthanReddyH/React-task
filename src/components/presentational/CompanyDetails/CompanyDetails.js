import React from "react";
import { Card, CardHeader, CardFooter, CardBody } from "reactstrap";
import "./CompanyDetails.scss";


const renderCards = employeeList => {
  return employeeList.map((item, index) => {
    return (
      <Card key={index} className="cardDetails">
        <CardHeader>{item.personName}</CardHeader>
        <CardBody>
          <label>
            <strong>Address : </strong>
          </label>{" "}
          {item.personAddress} <br />
        </CardBody>
        <CardFooter />
      </Card>
    );
  });
}

const CompanyDetails = (props) => {
  const { company } = props.location.state;

  return (
    <div className="ccontainer">
      <Card>
        <CardHeader>Profile Overview</CardHeader>
        <CardBody>
          <label>
            <strong>Address : </strong>
          </label>{" "}
          {company.address} <br />
          <label>
            <strong>Revnue : </strong>
          </label>{" "}
          {company.revenue}
          <br />
          <label>
            <strong>Phone : </strong>
          </label>{" "}
          {company.phone} <br />
          <label>
            <strong>No of Employees : </strong>
          </label>{" "}
          {company.employees.length}
        </CardBody>
        <CardFooter />
      </Card>
      { company.employees.length > 0 && <Card>
        <CardHeader>Employees</CardHeader>
        {renderCards(company.employees)}
      </Card> }
    </div>
  );
}

export default CompanyDetails;
