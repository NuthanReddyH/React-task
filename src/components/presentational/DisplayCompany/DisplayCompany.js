import React from "react";
import { Card, CardHeader, CardFooter, CardBody } from "reactstrap";
import { Link } from "react-router-dom";
import './DisplayCompany.scss'

const renderCards = companies => {
  return companies.map((item, index) => {
    return (
      <Card key={index} className="cardDetails">
        <CardHeader>{item.companyName}</CardHeader>
        <CardBody>
          <label>
            <strong>Address : </strong>
          </label>{" "}
          {item.address} <br />
          <label>
            <strong>Revnue : </strong>
          </label>{" "}
          {item.revenue}
          <br />
          <label>
            <strong>Phone : </strong>
          </label>{" "}
          {item.phone}
        </CardBody>
        <CardFooter>
          <Link to={{ pathname: "/details", state: { company: item } }}>
            Company Overview
          </Link>
        </CardFooter>
      </Card>
    );
  });
};
const DisplayCompany = ({ companies }) => {
  let message = "Currently there are no companies to review";
  if (!companies) {
    return <div className="errorMessage">{message}</div>;
  }
  let comp = renderCards(companies);
  return (
    <div className="ccontainer">
      <Card>
        <CardHeader>Companies</CardHeader>
        {comp}
      </Card>
    </div>
  );
};

export default DisplayCompany;
