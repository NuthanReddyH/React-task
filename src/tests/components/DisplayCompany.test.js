import React from "react";
import { shallow } from "enzyme";
import { Link } from 'react-router-dom';
import { MemoryRouter } from "react-router";
import DisplayCompany from '../../components/presentational/DisplayCompany/DisplayCompany';
import CompanyDetails from '../../components/presentational/CompanyDetails/CompanyDetails';

describe("Display Company component", () => {
    const companies = [
        {
            companyName: 'Amazon',
            address: '410 Terry Ave Seattle,WA',
            revenue: '100000',
            employees: [],
            phone: '(022)-266-0000'
        }
    ]
    const instance = shallow(<DisplayCompany companies={companies} />);
    it("should match the snapshot", () => {
      expect(instance).toMatchSnapshot();
    });


    
  });
  