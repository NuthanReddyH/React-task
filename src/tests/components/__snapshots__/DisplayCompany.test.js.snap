// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Company component should match the snapshot 1`] = `ShallowWrapper {}`;
